// @wordpress/block-editor/build/components/publish-date-time-picker/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _element = require("@wordpress/element");

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _components = require("@wordpress/components");

var _icons = require("@wordpress/icons");

var _i18n = require("@wordpress/i18n");

/**
 * WordPress dependencies
 */
function PublishDateTimePicker(_ref, ref) {
  let {
    onClose,
    onChange,
    ...additionalProps
  } = _ref;
  return (0, _element.createElement)("div", {
    ref: ref,
    className: "block-editor-publish-date-time-picker"
  }, (0, _element.createElement)(_components.__experimentalHStack, {
    className: "block-editor-publish-date-time-picker__header"
  }, (0, _element.createElement)("h2", {
    className: "block-editor-publish-date-time-picker__heading"
  }, (0, _i18n.__)('Publish')), (0, _element.createElement)(_components.__experimentalSpacer, null), (0, _element.createElement)(_components.Button, {
    className: "block-editor-publish-date-time-picker__reset",
    variant: "tertiary",
    onClick: () => onChange === null || onChange === void 0 ? void 0 : onChange(null)
  }, (0, _i18n.__)('Now')), (0, _element.createElement)(_components.Button, {
    className: "block-editor-publish-date-time-picker__close",
    icon: _icons.closeSmall,
    label: (0, _i18n.__)('Close'),
    onClick: onClose
  })), (0, _element.createElement)(_components.DateTimePicker, (0, _extends2.default)({
    __nextRemoveHelpButton: true,
    __nextRemoveResetButton: true,
    onChange: onChange
  }, additionalProps)));
}

var _default = (0, _element.forwardRef)(PublishDateTimePicker);

exports.default = _default;
