// @wordpress/block-editor/build-module/components/publish-date-time-picker/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { createElement } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { DateTimePicker, __experimentalHStack as HStack, __experimentalSpacer as Spacer, Button } from '@wordpress/components';
import { closeSmall } from '@wordpress/icons';
import { __ } from '@wordpress/i18n';
import { forwardRef } from '@wordpress/element';

function PublishDateTimePicker(_ref, ref) {
  let {
    onClose,
    onChange,
    ...additionalProps
  } = _ref;
  return createElement("div", {
    ref: ref,
    className: "block-editor-publish-date-time-picker"
  }, createElement(HStack, {
    className: "block-editor-publish-date-time-picker__header"
  }, createElement("h2", {
    className: "block-editor-publish-date-time-picker__heading"
  }, __('Publish')), createElement(Spacer, null), createElement(Button, {
    className: "block-editor-publish-date-time-picker__reset",
    variant: "tertiary",
    onClick: () => onChange === null || onChange === void 0 ? void 0 : onChange(null)
  }, __('Now')), createElement(Button, {
    className: "block-editor-publish-date-time-picker__close",
    icon: closeSmall,
    label: __('Close'),
    onClick: onClose
  })), createElement(DateTimePicker, _extends({
    __nextRemoveHelpButton: true,
    __nextRemoveResetButton: true,
    onChange: onChange
  }, additionalProps)));
}

export default forwardRef(PublishDateTimePicker);
